app/celery/nightly_tasks.py

```diff
@@ -122,6 +123,24 @@ def delete_letter_notifications_older_than_retention():
         raise
 
 
+@notify_celery.task(name="clean-notifications-table-and-create-stats")
+@statsd(namespace="tasks")
+def clean_notifications_table_and_create_stats():
+    """
+    A group of functions to run sequentially as their order matters
+
+    We timeout any notifications that are still in 'sending'. When that is done we can then create
+    the rows in the notification_status table using the data from the notifications table.
```

Comment:
does the solution also need to apply to `create_nightly_billing`? it's coded the same way as `create_nightly_notification_status` (ie: spawning more async tasks).

It doesn't depend on notifications having timed out, but if the delete task runs part way through it, then there could be issues as it could miss notifications that get moved to notification history while it's running